packages/apollo-gateway/src/index.ts

```diff
@@ -300,10 +300,18 @@ export class ApolloGateway implements GraphQLService {
   }
 
   public async load(options?: { engine?: GraphQLServiceEngineConfig }) {
-    await this.updateComposition(options);
+    if (options && options.engine) {
+      if (!options.engine.graphVariant)
+        this.logger.warn('No graph variant provided. Defaulting to `current`.');
+      this.engineConfig = options.engine;
+    }
 
-    if (isManagedConfig(this.config) || this.experimental_pollInterval) {
-      if (!this.pollingTimer) this.pollServices();
+    await this.updateComposition();
```

Comment:
I'm not sure that this PR accomplishes its goal. If this call throws, no polling will happen.

Reply:
Also note that even once this is fixed, there are a couple issues to resolve:
- We need special logic to ensure that the serverWillStart plugins get called, if they got skipped because the original schemaDerivedData threw
- The `Gateway successfully loaded schema` message in `load` won’t ever show up if the first update failed.  It might make sense to move that log line into `updateComposition` from `load`, eg putting it between assigning to `this.schema` and notifying the listeners, if `!previousSchema`